lib/value/src/value.rs

```diff
@@ -769,6 +770,32 @@ impl Value {
         }
     }
 
+    /// Returns a reference to a field value specified by a path iter.
+    pub fn get_by_path_v2<'a>(&self, path: impl Path<'a>) -> Option<&Self> {
```

Comment:
Should we name this (and others) `get_by_path_fast`, to make it more clear that this is the preferred option if one can accept the missing feature set?

Reply:
I think "v2" already makes it pretty clear this is the preferred option. Regardless, this is a fairly arbitrary name and will hopefully be renamed to just `get_by_path` once the old one is removed

Reply:
Fair point (although I'm wary of "once the old one is removed", given that we've had two implementations for two(?) years now).

But yeah, I'm cautiously optimistic that we can quickly schedule the work to upgrade the rest of the code base as well. I'll mark this as resolved.